### PR TITLE
Added error check for bad login credentials

### PR DESCRIPTION
**Pull request by xmclark**
Login method will write the status code and the status message to the error object if the login fails and the response returns a 200 status. 
Added okc_api=1 to the login request.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -7,8 +7,21 @@ var OKCupid = function(){
...
-    var login_form = {username: username, password: password};
-    this.requester.postRequest(URLS.login, login_form, callback)
+    var login_form = {
+      okc_api: 1, 
+      username: username, 
+      password: password,
+    };
+    this.requester.postRequest(URLS.login, login_form, function(err, res, body) {
+      // if there is a non-zero status in response and response does not show an error
+      if (body.status !== 0 && !err) {
```
Comment (tranhungt) [inline lib/client.js]: Should you check for `!err` first before checking `body.status`?
`!err && body.status !== 0`

Allows the logic to short circuit on error, and that you don't get unknown type error if `body` is undefined`.

**Comment by tranhungt** [comment]
Looks good. Thanks @xmclark.
Just check the little comment I made.

File: lib/client.js (hunk 1 above)
Comment (xmclark) [inline lib/client.js]: Ahh great catch. This is gotcha that would have taken me a little while to figure out. Thank you for the clarification. I just re-read the section on short-circuit evaluation on MDN. I will commit the change here shortly.

**Comment by tranhungt** [comment]
Looks good. Thanks @xmclark

**Pull request merged**